src/rules/no-confusing-set-timeout.ts

```diff
@@ -0,0 +1,66 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { createRule, getNodeName, parseJestFnCall } from './utils';
+
+function isJestSetTimeout(node: TSESTree.Node) {
+  return getNodeName(node) === 'jest.setTimeout';
+}
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Disallow using confusing setTimeout in test',
+      recommended: false,
+    },
+    messages: {
+      globalSetTimeout: '`jest.setTimeout` should be call in `global` scope.',
+      multipleSetTimeouts:
+        'Do not call `jest.setTimeout` multiple times, as only the last call will have an effect.',
+      orderSetTimeout:
+        '`jest.setTimeout` should be placed before any other jest methods',
+    },
+    type: 'problem',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    let seenJestTimeout = false;
+    let shouldEmitOrderSetTimeout = false;
+
+    return {
+      CallExpression(node) {
+        const scope = context.getScope();
+        const jestFnCall = parseJestFnCall(node, context);
+
+        if (!jestFnCall) {
+          return;
+        }
+
+        const result = isJestSetTimeout(node);
+
+        if (!result) {
+          if (jestFnCall.type !== 'unknown') {
+            shouldEmitOrderSetTimeout = true;
+          }
```

Comment:
By definition there is no known Jest function call of type `unknown` - it only exists to gracefully handle if Jest ever did get a new type in older versions of the plugin (since we'd have to release a new version adding support); so you shouldn't worry about it here:

```suggestion
          shouldEmitOrderSetTimeout = true;
```